apps/telegram_bot/main.py

```diff
@@ -0,0 +1,348 @@
+        # Accept text or voice messages
+        if update.message and update.message.voice:
+            user_telegram_voice = await context.bot.get_file(
+                update.message.voice.file_id
+            )
+            bytes = await user_telegram_voice.download_as_bytearray()
+            # convert audio bytes to numpy array
+            input = AudioSegment.from_file(
+                io.BytesIO(bytes), format="ogg", codec="libopus"  # type: ignore
+            )
+        elif update.message and update.message.text:
+            input = update.message.text
+        else:
+            # No audio or text, complain to user.
+            await context.bot.send_message(
+                chat_id=update.effective_chat.id,
+                text="""
+Sorry, I only respond to commands, voice, or text messages. Use /help for more information.""",
+            )
+            return
+
+        # Get audio response from LLM/synth and reply
+        agent_response, synth_response = await self.get_response(int(chat_id), input)
+        out_voice = io.BytesIO()
+        synth_response.export(out_f=out_voice, format="ogg", codec="libopus")  # type: ignore
+        await context.bot.send_message(
+            chat_id=update.effective_chat.id, text=agent_response
+        )
+        await context.bot.send_voice(chat_id=str(chat_id), voice=out_voice)
+
+    async def handle_telegram_select_voice(
+        self, update: Update, context: ContextTypes.DEFAULT_TYPE
+    ) -> None:
+        assert update.effective_chat, "Chat must be defined!"
+        chat_id = update.effective_chat.id
+        if not (context.args):
+            await context.bot.send_message(
+                chat_id=chat_id,
+                text="You must include a voice id. Use /list to list available voices",
+            )
+            return
+        new_voice_id = context.args[0]
+
+        user_voices = self.db[chat_id]["voices"]
+        if len(user_voices) <= int(new_voice_id):
+            await context.bot.send_message(
+                chat_id=chat_id,
+                text="Sorry, I do not recognize that voice. Use /list to list available voices.",
+            )
+            return
+        else:
+            self.db[chat_id]["current_voice"] = user_voices[new_voice_id]
+            # Reset conversation
+            self.db[chat_id]["current_conversation"] = None
+            await context.bot.send_message(
+                chat_id=chat_id, text="Voice changed successfully!"
+            )
+
+    async def handle_telegram_create_voice(
+        self, update: Update, context: ContextTypes.DEFAULT_TYPE
+    ) -> None:
+        assert update.effective_chat, "Chat must be defined!"
+        chat_id = update.effective_chat.id
+        if type(self.synthesizer) is not CoquiSynthesizer:
+            await context.bot.send_message(
+                chat_id=chat_id,
+                text="Sorry, voice creation is only supported for Coqui TTS.",
```

Comment:
```suggestion
                text="Sorry, voice creation is only supported for Coqui.",
```

since we have a "Coqui TTS" synthesizer which is their OSS thing

Reply:
fixed

Reply:
don't see the change